### PR TITLE
🐛Fix a2a for next article links in amp-next-page

### LINKED ISSUE #18512: Cannot read `navigateToAmpUrl` of null
`this.navigation_` is always null when read here:

https://github.com/ampproject/amphtml/blob/a348bbc5f2e4e3640ac4f84afadc6ad9e7de5f43/extensions/amp-next-page/0.1/next-page-service.js#L317-L318

The private field is declared here:

https://github.com/ampproject/amphtml/blob/a348bbc5f2e4e3640ac4f84afadc6ad9e7de5f43/extensions/amp-next-page/0.1/next-page-service.js#L91-L92

But the field is never initialized in the file. This error can be reproduced clicking on a link within an `<amp-next-page>`.

Since this error happens in a click listener, it will cause other click listeners for the event to not be invoked.

This was broken in #17600

### PR DESCRIPTION
**Pull request by sparhami**
Fixes #18512

### CODE REVIEW

**Comment by tchirag** [comment]
Is this removal of Analytics trigger on amp-next-page?

**Comment by sparhami** [comment]
> Is this removal of Analytics trigger on amp-next-page?

I believe that the event that triggers for a2a within `<amp-next-page>` would have triggered before this fix as well, since the code for the event happens before the line with the error. The actual amp to amp navigation would not have happened though.

**Comment by tchirag** [comment]
> > Is this removal of Analytics trigger on amp-next-page?
> 
> I believe that the event that triggers for a2a within `<amp-next-page>` would have triggered before this fix as well, since the code for the event happens before the line with the error. The actual amp to amp navigation would not have happened though.

Unfortunately with this push, trigger for pageview event on Google Analytics has stopped on amp-next-page. To explain, pageview event gets triggered on first page however doesnt get triggered on 2nd and 3rd page which are loaded using amp-next-page. Is this expected outcome?

File: extensions/amp-next-page/0.1/next-page-service.js [hunk 1]
```diff
@@ -138,6 +138,7 @@ export class NextPageService {
       this.hideSelector_ = this.config_.hideSelectors.join(',');
     }
 
+    this.navigation_ = Services.navigationForDoc(ampDoc);
     this.viewport_ = Services.viewportForDoc(ampDoc);
```
Comment (123buyDoMiNiK) [inline extensions/amp-next-page/0.1/next-page-service.js]: fawsaegvsedgvsaegsegbsegbvsehdtk,ghy.lugc,dxrtgewsa4waq4YHRSj4

Comment (123buyDoMiNiK) [reply extensions/amp-next-page/0.1/next-page-service.js]: KILOHASJGRNBVGSBVIUJSOPFDQKAF[LQ[PFJWIOFGIUWGFPQ[LDC';AFVCMLSABHGVEWIFKJWQ[APFPOQWAGIUREWSFPA[WLKFIOGURHWWP[GFKOPESJGVOPSEHGVIOASEJHPOGESAOLFJVOIAWGHFAWQP[QFKAEP[SEHSGVAJOP;FVKQA[PQGFPOWAUQEFO9IQUY3W09FGIQ=-3RF903476J2

Comment (123buyDoMiNiK) [reply extensions/amp-next-page/0.1/next-page-service.js]: IOJGRES09OGV[EPWSAOL[PELSA[]GLEGL[PSELKGP[KSP[KGDLVG;'P[SEKOPGFOPSEJIFUEAYOIJ[PWFK[PAKWFIW

Comment (123buyDoMiNiK) [reply extensions/amp-next-page/0.1/next-page-service.js]: OFNDJIKVFJHDEGHJKEGHJKIUSHIOGHJISOEHGIOSEHGIOSEJGIOEJGIOBGVBBFBBXBKBRNFIOJAIOFJAQOIFJIOAQHFIOUQYHF89QWYAFGIOHJWQIOGFJWOJGFOPWJGFPWJQKAGPFOJWSOAIGVJWEOISAEGJPOWSEGJOPWAJEGOPWSEJGOPWEJSGOPIJEWS

**Comment by sparhami** [comment]
> Unfortunately with this push, trigger for pageview event on Google Analytics has stopped on amp-next-page. To explain, pageview event gets triggered on first page however doesnt get triggered on 2nd and 3rd page which are loaded using amp-next-page. Is this expected outcome?

Do you mean an analytics event when the user scrolls and the next page is loaded or when the user clicks on a link (in the [recommendation box](https://www.ampproject.org/docs/reference/components/amp-next-page#recommendation-box). To the best of my knowledge, this change should only affect when the user clicks on a link. Are you seeing the missing analytics when you click on one of those links?

@peterjosling do you have any insight as to what may be going on? I'm not familiar with the amp 2 amp navigation flow, could there be some problem there?

**Comment by sparhami** [comment]
Hmm, I wonder if this is related to the fact that we strip the `<amp-analytics>` config from `<amp-next-page>`. It may be that the analytics was working before because it was a hard page navigation. Now, the event fires to let it know that an a2a navigation happened, then we do an a2a navigation. It could be that nothing is actually listening for that event though.

**Comment by tchirag** [comment]
Analytics event doesnt get triggered when the user scrolls and the next
page is loaded. I.e. Earlier, Upon loading of next page a separate page
event was getting triggered which is not happening now. Could you pls
look into this?

**Comment by sparhami** [comment]
@lannka Are you aware of any issues or changes with `<amp-analytics>` and/or `<amp-next-page>` that might prevent the analytics triggering?

**Comment by peterjosling** [comment]
Analytics are currently disabled in child documents. https://github.com/ampproject/amphtml/issues/15807

**Comment by tchirag** [comment]
Even creating DFP Ad Units isnt possible currently?

**Pull request merged**